docs/logicapps-ai-course/04_build_multi_agent_systems/02_nested_agents.md

```diff
@@ -0,0 +1,134 @@
+# Module 02 - Use nested agents as tools
+In this module, we'll use the Logic Apps Nested Agent action to build a multi-agent system for publishing a monthly sales report. The Nested Agent action leverages the agent-as-tool framework.
+
+## Agent-as-Tool Framework
+
+In a multi-agent system, using the agent-as-tool framework allows one agent to delegate a complex task to another specialized agent. This approach offers several key advantages:
+
+- **Modularity**: Each nested agent focuses on a specific, complex task, making the overall system easier to design and manage
+- **Reusability**: You can reuse the same specialized nested agent across different workflows, saving development time and effort
+- **Context preservation**: Nested agents can maintain their own conversation history and state, enabling them to produce more nuanced, context-aware outputs
+
+## When to use the Nested Agent action?
+
+Use the Logic Apps Nested Agent action when a supervisor agent needs to delegate a complex subtask that requires a specialized, multi-step process to a child agent. The supervisor remains in control of the overall workflow and integrates the child agent's output to complete the final task.
+
```

Comment:
can we add a ## Course Outline section similar to the one that Abode and Edward have in their PRs?